src/core/ClearStack.js

```diff
@@ -70,7 +70,9 @@ getJasmineRequireObj().clearStack = function(j$) {
 
     const SAFARI =
       global.navigator &&
-      /^((?!chrome|android).)*safari/i.test(global.navigator.userAgent);
+      /(^((?!chrome|android).)*safari)|(^((?!chrome|android|firefox).)+$)/i.test(
+        global.navigator.userAgent
+      );
 
     if (NODE_JS) {
```

Comment:
> Is there more detail on why/if running tests on browsers really can't use the more lightweight nodeQueueMicrotaskImpl?

The short answer is that we've tried not doing a setTimeout in browsers several times and it ended badly each time. https://github.com/jasmine/jasmine/issues/2008 was the last attempt. There are a few reasons:

* Browsers don't allow JS to run forever without yielding control. When a site does that, most browsers will eventually decide that it's caught in an infinite loop and kill the tab.
* Even if they did, tools like Karma conclude that the test runner has crashed if it goes too long without producing any output.
* Some browsers will defer or minimize garbage collection if the running code doesn't go idle enough. This is probably why previous attempts to optimize this have caused major slowdowns for suites with a lot of synchronous specs. 

Those problems mainly show up in suites with a lot of synchronous specs, or even a few heavyweight synchronous specs that happen to get scheduled back to back. You won't see them if all of your specs contain async or if you're only running a few focused specs.

Reply:
Has there been any attempt at an elapsed time based approach, rather than something like "every 10 calls"? I.e. only call setTimeout if it has been at least 500ms since the last setTimeout. I can't imagine something like that would significantly increase the chance of getting killed by the browser, and it would probably be a big win when there are lots of short or skipped tests. I'd be happy to put up a PR for something like that if there's interest.

Reply:
Yes, and it dramatically slowed down mostly-synchronous test suites including jasmine-core's. It could also be enough to cause some test suites to run out of memory.

More generally, getting this right is _hard_. It took years to stabilize the between-specs stack clearing process so that nobody's test suite crashed their browser, crashed Karma, ran out of memory or ran unreasonably slow. Those are the worst kind of bugs: hard to find before release, completely breaks Jasmine for some people, and the affected users often can't provide steps to reproduce the problem. 

The existing approach looks crude but it's been running since 2017 with no complaints except for this one and the issue I linked to above. I'm not completely unwilling to change it, but the bar to demonstrating that a change is safe is quite high. We'd need reason to be confident that the change wouldn't break Jasmine for anyone, regardless of what their test suite is shaped like. That's hard to come by.